verilog/analysis/checkers/declared_identifier_style_rule.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_ANALYSIS_CHECKERS_DECLARED_IDENTIFIER_STYLE_RULE_H_
+#define VERIBLE_VERILOG_ANALYSIS_CHECKERS_DECLARED_IDENTIFIER_STYLE_RULE_H_
+
+#include <set>
+#include <string>
+
+#include "common/analysis/lint_rule_status.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/analysis/matcher/matcher_builders.h"
+#include "common/analysis/syntax_tree_lint_rule.h"
+#include "common/text/symbol.h"
+#include "common/text/syntax_tree_context.h"
+#include "verilog/CST/verilog_matchers.h"  // IWYU pragma: keep
+#include "verilog/analysis/descriptions.h"
+
+namespace verilog {
+namespace analysis {
+
+// checks declared identifier with set of unwanted patterns
+class DeclaredIdentifierStyleRule : public verible::SyntaxTreeLintRule {
```

Comment:
Can we name this class `BannedDeclaredNamePatternsRule`?  (same with the file name?)

Reply:
Done!